mongo/integration/mtest/mongotest.go

```diff
@@ -726,6 +727,77 @@ func (t *T) createTestClient() {
 	}
 }
 
+func (t *T) testCommandInterceptor() {
```

Comment:
I couldn't get the tests to actually run locally. But threw this together as a simple attempt to show how it might be tested